src/main/java/sparksoniq/io/shell/JiqsJLineShell.java

```diff
@@ -118,8 +120,11 @@ private void initialize() throws IOException {
     }
 
     private void handleException(Exception ex) {
-        if (ex != null && !(ex instanceof UserInterruptException))
+        if (ex != null && ex instanceof EndOfFileException) {
```

Comment:
Use an outside if for the null check and 2 nested ifs or a switch for the instance check.